src/windowsdesktop/src/windowsdesktop/src/bundle/Product.targets

```diff
@@ -0,0 +1,100 @@
+<Project>
+  <PropertyGroup>
+    <DotNetDarkOutputPath>$(IntermediateOutputPath)dark\</DotNetDarkOutputPath>
+    <DepsPath>$(BaseIntermediateOutputPath)</DepsPath>
+    <DefineConstants>$(DefineConstants);DepsPath=$(DepsPath)</DefineConstants>
+    <DefineConstants>$(DefineConstants);InstallersOutputPath=$(InstallersOutputPath)</DefineConstants>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(TargetArchitecture)' == 'x86'">
+    <RuntimeInstallers Include="$(DepsPath)dotnet-runtime-$(MicrosoftNETCoreAppRefVersion)-win-x86.msi">
+      <BundleNameProperty>DotNetRedistLtsInstaller</BundleNameProperty>
+    </RuntimeInstallers>
+    <RuntimeInstallers Include="$(DepsPath)dotnet-host-$(MicrosoftNETCoreAppRefVersion)-win-x86.msi">
+      <BundleNameProperty>DotNetRedistHostInstaller</BundleNameProperty>
+    </RuntimeInstallers>
+    <RuntimeInstallers Include="$(DepsPath)dotnet-hostfxr-$(MicrosoftNETCoreAppRefVersion)-win-x86.msi">
+      <BundleNameProperty>DotNetRedistHostfxrInstaller</BundleNameProperty>
+    </RuntimeInstallers>
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetArchitecture)' == 'x64'">
+    <RuntimeInstallers Include="$(DepsPath)dotnet-runtime-$(MicrosoftNETCoreAppRefVersion)-win-x64.msi">
+      <BundleNameProperty>DotNetRedistLtsInstaller</BundleNameProperty>
+    </RuntimeInstallers>
+    <RuntimeInstallers Include="$(DepsPath)dotnet-host-$(MicrosoftNETCoreAppRefVersion)-win-x64.msi">
+      <BundleNameProperty>DotNetRedistHostInstaller</BundleNameProperty>
+    </RuntimeInstallers>
+    <RuntimeInstallers Include="$(DepsPath)dotnet-hostfxr-$(MicrosoftNETCoreAppRefVersion)-win-x64.msi">
+      <BundleNameProperty>DotNetRedistHostfxrInstaller</BundleNameProperty>
+    </RuntimeInstallers>
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(TargetArchitecture)' == 'arm64'">
+    <RuntimeInstallers Include="$(DepsPath)dotnet-runtime-$(MicrosoftNETCoreAppRefVersion)-win-arm64.msi">
+      <BundleNameProperty>DotNetRedistLtsInstaller</BundleNameProperty>
+    </RuntimeInstallers>
+    <RuntimeInstallers Include="$(DepsPath)dotnet-host-$(MicrosoftNETCoreAppRefVersion)-win-arm64.msi">
+      <BundleNameProperty>DotNetRedistHostInstaller</BundleNameProperty>
+    </RuntimeInstallers>
+    <RuntimeInstallers Include="$(DepsPath)dotnet-hostfxr-$(MicrosoftNETCoreAppRefVersion)-win-arm64.msi">
+      <BundleNameProperty>DotNetRedistHostfxrInstaller</BundleNameProperty>
+    </RuntimeInstallers>
+  </ItemGroup>
+
+  <!--
+    Runs before FetchDependencies but can't be depended up because this sets up the item group that target
+    uses for batching.
+  -->
+  <Target Name="CollectDependencies" BeforeTargets="Restore;CollectPackageReferences">
+    <!-- Use the BrowserDebugHost as a sentinel for the nonshipping version for NETCoreApp. -->
+    <ItemGroup>
+      <RemoteAsset Include="$(MicrosoftNETCorePlatformsVersion)/dotnet-runtime-$(MicrosoftNETCoreAppRefVersion)-win-$(TargetArchitecture).msi">
```

Comment:
nit: I don't think you need this `ItermGroup` at all. You could just replace its uses with the `RuntimeInstallers` ItemGroup (some extra metadata may be needed)